### PR TITLE
add to inspection to send messages upon completion

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

#398 
https://github.com/thoth-station/workflow-helpers/pull/198

## This introduces a breaking change

- [ ] Yes
- [x] No

## Test or Stage Environment

- [ ] Pull Requests added content to STAGE environment
- [ ] Pull Requests to `AICoE/aicoe-cd` has been opened: 

## This Pull Request implements

This will send a message which will be handled by investigator to schedule graph-sync after a inspection completes

### CODE REVIEW

File: amun/base/kustomization.yaml [hunk 1]
```diff
@@ -9,6 +9,9 @@ resources:
   - argo-workflows/inspection-build-with-cpu.yaml
   - argo-workflows/inspection-run.yaml
   - argo-workflows/inspection-run-with-cpu.yaml
+  - argo-workflows/parse-inspection-inputs.yaml
+  - ../../../core/base/argo-workflows/send-message.yaml
```
Comment (pacospace) [inline amun/base/kustomization.yaml:13]: I think we want amun to be independent component, therefore send-message.yaml template need to be place under amun repo and not reference core. cc @harshad16

File: amun/base/kustomization.yaml [hunk 2]
```diff
@@ -9,6 +9,9 @@ resources:
...
+  - ../../../core/base/argo-workflows/send-messages.yaml
```
Comment (pacospace) [inline amun/base/kustomization.yaml:14]: same for template above

File: amun/base/kustomization.yaml (hunk 1 above)
Comment (harshad16) [reply amun/base/kustomization.yaml:13]: yes, as @pacospace stated, we want amun to be independent component , so please move all amun related manifest under the amun directory.

Comment (KPostOffice) [reply amun/base/kustomization.yaml:13]: @harshad16 @pacospace Is it okay to use a symlink here?

Comment (harshad16) [reply amun/base/kustomization.yaml:13]: I know it is not convenient to maintain two copies, but I feel we should do it in this case.
Using symlink won't solve the independence we require for Amun.

File: amun/base/kustomization.yaml [hunk 3]
```diff
@@ -9,6 +9,9 @@ resources:
   - argo-workflows/inspection-build-with-cpu.yaml
   - argo-workflows/inspection-run.yaml
   - argo-workflows/inspection-run-with-cpu.yaml
+  - argo-workflows/parse-inspection-inputs.yaml
+  - argo-workflows/send-message.yaml
```
Comment (pacospace) [inline amun/base/kustomization.yaml:13]: do we need this template here? I see only `send-messages.yaml` is used in inspection workflow templates

File: amun/base/imagestream.yaml [hunk 4]
```diff
@@ -22,3 +22,19 @@ metadata:
...
+  lookupPolicy:
+    local: true
+---
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: messaging
+spec:
+  lookupPolicy:
+    local: true
```
Comment (pacospace) [inline amun/base/imagestream.yaml:40]: you also need to create one `imagestreamtag.yaml` in the different overlays and reference them in the `kustomize.yaml` file in respective overlays, otherwise no specific image is used. for example: https://github.com/thoth-station/thoth-application/blob/master/amun/overlays/ocp4-stage/amun-api/imagestreamtag.yaml

Comment (KPostOffice) [reply amun/base/imagestream.yaml:40]: This is only necessary in `prod`, `stage`, and `ocp4-stage`? (any that already have `imagestreamtag.yaml` defined in the overlay)

Comment (KPostOffice) [reply amun/base/imagestream.yaml:40]: Nevermind, I realize that `power9` is the only overlay with an `imagestreamtag.yaml`

**Comment by goern** [comment]
how is this one doing?!

File: amun/overlays/cnv-prod/amun-api/imagestreamtag.yaml [hunk 5]
```diff
@@ -39,3 +39,28 @@ spec:
...
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: workflow-helpers
+spec:
+  tags:
+  - name: latest
+    from:
+      kind: DockerImage
+      name: quay.io/thoth-station/workflow-helpers:latest
```
Comment (pacospace) [inline amun/overlays/cnv-prod/amun-api/imagestreamtag.yaml:52]: we should not use the `latest` tag as we do not know what is that from here, but have a specific tag here for our components for easily debug and make sure that we have the version we want.

Same for messaging below.

File: amun/overlays/ocp4-stage/amun-api/imagestreamtag.yaml [hunk 6]
```diff
@@ -39,3 +39,28 @@ spec:
...
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: workflow-helpers
+spec:
+  tags:
+  - name: latest
+    from:
+      kind: DockerImage
+      name: quay.io/thoth-station/workflow-helpers:latest
```
Comment (pacospace) [inline amun/overlays/ocp4-stage/amun-api/imagestreamtag.yaml:52]: Same comments as above related to not use `latest`

**Review by pacospace — COMMENTED** [review]
One more comment and it should be ready! Thanks! 💯

**Comment by KPostOffice** [comment]
@pacospace PTAL

**Comment by pacospace** [comment]
> @pacospace PTAL

Looks good to me!! Did we tested in test environment?

**Comment by KPostOffice** [comment]
@pacospace I will test all new components in the test env

File: amun/base/openshift-templates/inspection-workflow.yaml [hunk 7]
```diff
@@ -115,12 +120,54 @@ objects:
...
   required: true
 
+- name: THOTH_SEND_MESSAGES
+  description: indicates whether message should be sent upon inspection completion
+  displayName: send messages
+  value: "1"  # sends message by default
+
+- name: THOTH_FORCE_SYNC
+  description: should inspection complete message cause graph sync job to be run with force sync
+  display_name: force sync
```
Comment (harshad16) [inline amun/base/openshift-templates/inspection-workflow.yaml:168]:
```suggestion
  displayName: force sync
```

File: amun/base/openshift-templates/inspection-with-cpu-workflow.yaml [hunk 8]
```diff
@@ -138,12 +143,55 @@ objects:
...
   required: true
 
+- name: THOTH_SEND_MESSAGES
+  description: indicates whether message should be sent upon inspection completion
+  displayName: send messages
+  value: "1"  # sends message by default
+
+- name: THOTH_FORCE_SYNC
+  description: should inspection complete message cause graph sync job to be run with force sync
+  display_name: force sync
```
Comment (harshad16) [inline amun/base/openshift-templates/inspection-with-cpu-workflow.yaml:192]:
```suggestion
  displayName: force sync
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
lgtm on the changes wise, I have added few suggestions
i would like to keep the prod changes out of this for now.

can you the openshift-templates folder to test and ocp4-stage
and add these files in respective kustomize file.
you can remove the changes from prod and stage overlays, we are not using them currently.

File: amun/overlays/ocp4-stage/amun-api/kustomization.yaml [hunk 9]
```diff
@@ -10,6 +10,8 @@ patchesStrategicMerge:
   - imagestreamtag.yaml
 generators:
   - ./secret-generator.yaml
+  - openshift-templates/inspection-workflow.yaml
+  - openshift-templates/inspection-with-cpu-workflow.yaml
```
Comment (harshad16) [inline amun/overlays/ocp4-stage/amun-api/kustomization.yaml:14]: This should go under the patchStrategyMerge field as this would update the template defined in the base.

**Review by harshad16 — CHANGES_REQUESTED** [review]
we can lose the `stage` overlay changes as that overlay doesnt get deployed.
Also requested some changes which would be critical for this PR.

**Comment by KPostOffice** [comment]
@harshad16 This should be good to go now

File: amun/base/openshift-templates/inspection-workflow.yaml [hunk 10]
```diff
@@ -115,20 +117,18 @@ objects:
...
             failed: true
 
-parameters:
-- name: AMUN_INSPECTION_ID
-  description: Id of inspection that is run.
-  displayName: Inspection id
-  required: true
+- name: THOTH_FORCE_SYNC
+  description: should inspection complete message cause graph sync job to be run with force sync
+  displayName: force sync
```
Comment (harshad16) [inline amun/base/openshift-templates/inspection-workflow.yaml:122]: These changes cant go in, as it would disrupt other environments.
i noticed this is already been add to ocp4 and test, so kindly revert the changes in file from base.

**Review by harshad16 — CHANGES_REQUESTED** [review]
lgtm,
suggested some changes.

File: amun/base/openshift-templates/inspection-workflow.yaml (hunk 10 above)
Comment (KPostOffice) [reply amun/base/openshift-templates/inspection-workflow.yaml:122]: Ah yes, sorry. Thanks for catching this Harshad

File: amun/base/kustomization.yaml [hunk 11]
```diff
@@ -9,6 +9,8 @@ resources:
   - argo-workflows/inspection-build-with-cpu.yaml
   - argo-workflows/inspection-run.yaml
   - argo-workflows/inspection-run-with-cpu.yaml
+  - argo-workflows/parse-inspection-inputs.yaml
```
Comment (harshad16) [inline amun/base/kustomization.yaml:12]:
```suggestion
  - argo-workflows/create-inspection-complete-message.yaml
```
there is no parse-inspection-inputs.yaml added or available

**Review by harshad16 — CHANGES_REQUESTED** [review]
some more changes required.

File: amun/base/openshift-templates/inspection-workflow.yaml [hunk 12]
```diff
@@ -127,7 +127,7 @@ parameters:
   value: 750m
 
 - name: AMUN_RUN_MEMORY
-  description: Memory requested for inspection run
+  description: Memory requsted for inspection run
```
Comment (harshad16) [inline amun/base/openshift-templates/inspection-workflow.yaml:130]: this can be reverted.

File: amun/base/kustomization.yaml [hunk 13]
```diff
@@ -9,8 +9,8 @@ resources:
   - argo-workflows/inspection-build-with-cpu.yaml
   - argo-workflows/inspection-run.yaml
   - argo-workflows/inspection-run-with-cpu.yaml
-  - openshift-templates/inspection-workflow.yaml
-  - openshift-templates/inspection-with-cpu-workflow.yaml
```
Comment (harshad16) [inline amun/base/kustomization.yaml:13]: removing these file will disrupt all the enviroments.

**Review by harshad16 — CHANGES_REQUESTED** [review]
eveyrthing else lgtm,
just one change need  to be reverted.

**Review by harshad16 — APPROVED** [review]
/approve
Thanks for these additions
This can now go in.
